test/zAuctionRouting.test.ts

```diff
@@ -13,46 +15,89 @@ describe("zAuctionRouting", () => {
+    // subgraphUri: "",
+    // zAuctionAddress: "",
+    // tokenContract: "",
+    // web3Provider: ethers.providers.getDefaultProvider(),
+  const dummyZAuctionConfig = {
+    apiUri: "http",
+    subgraphUri: "http",
+    zAuctionAddress: "0x1",
+    tokenContract: "http",
+    web3Provider: ethers.providers.getDefaultProvider(),
+  }
+
+  const dummyConfig = {
+    subgraphUri: "http",
+    metricsUri: "http",
+    apiUri: "http",
+    zAuctionRoutes: [],
+    basicController: "http",
+    registrar: "http",
+  } as Config
+
+  createZAuctionInstances(dummyConfig);
+
+  const dummyInstance = {
+    listSales: () => {},
+    listBids: (tokenIds: string[]) => {},
+    listBidsByAccount: (account: string) => {},
+    placeBid: (params: zAuction.NewBidParameters, signer: ethers.Signer, statusCallback?: zAuction.PlaceBidStatusCallback) => {},
+    isZAuctionApprovedToTransferNft: (account: string) => {},
+    getZAuctionSpendAllowance: (account: string) => {},
+    getTradeTokenAddress: () => {},
+    approveZAuctionSpendTradeTokens: (signer: ethers.Signer) => {},
+    approveZAuctionTransferNft: (signer: ethers.Signer) => {},
+    acceptBid: (bid: zAuction.Bid, signer: ethers.Signer) => {}
+  } as unknown as zAuction.Instance;
 
   it("Returns the proper instance", async () => {
     const routes = [
       {
         uriPattern: "wilder",
-        instance: dummyInstance1,
+        config: dummyZAuctionConfig as zAuction.Config,
       } as zAuctionRoute,
       {
         uriPattern: "",
-        instance: dummyInstance2,
+        config: dummyZAuctionConfig as zAuction.Config,
       } as zAuctionRoute,
     ];
 
-    const instance = await getZAuctionInstanceForDomain(
+    dummyConfig.zAuctionRoutes = routes;
+    const zAuctionRouteUriToInstance = createZAuctionInstances(dummyConfig);
+
+    const instance: zAuction.Instance = await getZAuctionInstanceForDomain(
       "0x1",
       routes,
+      zAuctionRouteUriToInstance,
       idToNameStub
     );
-    expect(instance).to.eq(dummyInstance1);
+    // const instance = zAuction.createInstance();
+    expect(JSON.stringify(instance)).to.eq(JSON.stringify(dummyInstance))
   });
 
   it("Returns default route", async () => {
     const routes = [
       {
         uriPattern: "wilder",
-        instance: dummyInstance1,
+        config: dummyZAuctionConfig,
       } as zAuctionRoute,
       {
         uriPattern: "",
-        instance: dummyInstance2,
+        config: dummyZAuctionConfig,
       } as zAuctionRoute,
     ];
 
-    const instance = await getZAuctionInstanceForDomain(
+    dummyConfig.zAuctionRoutes = routes;
+    const zAuctionRouteUriToInstance = createZAuctionInstances(dummyConfig);
+
+    const instance: zAuction.Instance = await getZAuctionInstanceForDomain(
       "0x2",
       routes,
+      zAuctionRouteUriToInstance,
       idToNameStub
     );
-    expect(instance).to.eq(dummyInstance2);
+    expect(JSON.stringify(instance)).to.eq(JSON.stringify(dummyInstance));
```

Comment:
super nit - if we use some sort of deep object equality here, we'll get better messaging if the objects don't turn out to be equal. Not gonna block on this one though

Reply:
Good callout, I've added a deeper equality check between the returned instance and our dummy instance functions